udmis/src/test/java/com/google/bos/udmi/service/core/StateHandlerTest.java

```diff
@@ -0,0 +1,146 @@
+package com.google.bos.udmi.service.core;
+
+import static com.google.bos.udmi.service.messaging.LocalMessagePipe.getQueueForScope;
+import static com.google.udmi.util.GeneralUtils.deepCopy;
+import static com.google.udmi.util.JsonUtil.fromString;
+import static com.google.udmi.util.JsonUtil.stringify;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static udmi.schema.Envelope.SubFolder.SYSTEM;
+import static udmi.schema.Envelope.SubType.STATE;
+
+import com.google.bos.udmi.service.messaging.LocalMessagePipe;
+import com.google.bos.udmi.service.messaging.LocalMessagePipeTest;
+import com.google.bos.udmi.service.messaging.MessageBase.Bundle;
+import com.google.bos.udmi.service.messaging.MessagePipe;
+import com.google.bos.udmi.service.messaging.MessageTestBase;
+import java.util.concurrent.BlockingQueue;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+import udmi.schema.Envelope;
+import udmi.schema.GatewayState;
+import udmi.schema.MessageConfiguration;
+import udmi.schema.MessageConfiguration.Transport;
+import udmi.schema.State;
+import udmi.schema.SystemState;
+
+/**
+ * Tests for the StateHandler class, used by UDMIS to process device state updates.
+ */
+public class StateHandlerTest extends MessageTestBase {
+
+  public static final String INVALID_MESSAGE = "invalid message";
+  private StateHandler stateHandler;
+
+  @Test
```

Comment:
All helper functions first then tests?

Reply:
As per other comment -- not something I ever pay attention to, so unless there's something automatic about detecting it I doubt it'll be a consistent convention. I can try to rearrange some, but it just seems like unnecessary overhead to me.